src/main/java/org/apache/mesos/hdfs/executor/NameNodeExecutor.java

```diff
@@ -99,13 +99,12 @@ public void frameworkMessage(ExecutorDriver driver, byte[] msg) {
     File nameDir = new File(schedulerConf.getDataDir() + "/name");
     if (messageStr.equals(HDFSConstants.NAME_NODE_INIT_MESSAGE)
         || messageStr.equals(HDFSConstants.NAME_NODE_BOOTSTRAP_MESSAGE)) {
-      if (nameDir.exists()) {
-        // TODO (elingg) test failure scenario to make sure NN restarts with the
-        // appropriate sequence of events
+      if (nameDir.exists() && messageStr.equals(HDFSConstants.NAME_NODE_INIT_MESSAGE)) {
```

Comment:
Yes, see my above comment. We only call this if another NN is running and active. You want the to bootstrap off of the latest data from the Active NN.